### PR TITLE
Perezavidien/asp net core masters/session4

### PR DESCRIPTION
**Pull request by perezavidien**
Apologies that I wasn't able to create a PR last Saturday

### CODE REVIEW

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs [hunk 1]
```diff
@@ -11,39 +11,72 @@
...
         {
-            return _itemService.GetAll(userId);
+            var result = _itemService.GetAll();
+            return Ok(result);
         }
 
         [HttpGet]
-        int Get(int userId)
+        [Route("{itemId}")]
+        IActionResult Get(int id)
```
Comment (jegtugado) [inline ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs:36]: parameter should match what's in the route template

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs [hunk 2]
```diff
@@ -11,39 +11,72 @@
 
 namespace ASPNetCoreMastersTodoList.Api.Controllers
 {
+    [ApiController]
+    [Route("items")]
```
[the remark this answers is not in the document]
Comment (perezavidien) [reply ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: thank you for reviewing, this was updated.

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs [hunk 3]
```diff
@@ -11,39 +11,72 @@
...
+        [Route("{itemId}")]
+        IActionResult Get(int id)
         {
-            return 1;
+            var result = _itemService.GetById(id);
+            return Ok(result);
+        }
+
+        [HttpGet]
+        [Route("filterBy?[text]=[text]")]
```
[the remark this answers is not in the document]
Comment (perezavidien) [reply ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: thank you for reviewing, this was updated.

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs (hunk 1 above)
Comment (perezavidien) [reply ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: thank you for reviewing, this was updated.

File: ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs [hunk 4]
```diff
@@ -11,39 +11,72 @@
...
+            var itemDto = new ItemDTO(itemCreateModel.Text);            
 
             _itemService.Save(itemDto);
+
+            return Ok();
+        }
+
+        [HttpPut]
+        [Route("{itemId}")]
+        IActionResult Put(int id, [FromBody] ItemUpdateBindingModel itemUpdateModel)
```
[the remark this answers is not in the document]
Comment (perezavidien) [reply ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs]: thank you for reviewing, this was updated.

**Comment by perezavidien** [comment]
thank you for reviewing, this was updated.

**Review by jegtugado — APPROVED** [review]

**Pull request merged**